limesurvey/limesurvey.py

```diff
@@ -91,6 +103,18 @@ def studio_submit(self, data, suffix=""):  # pylint: disable=unused-argument
 
         return {"result": "success"}
 
+    @XBlock.json_handler
+    def get_survey_url(self, data, suffix=''): # pylint: disable=unused-argument
+        """
+        Show the survey URL and access code to the user
+        """
+        URL = "http://limesurvey.local.overhang.io:8082"
+
+        self.survey_url = f"{URL}/{self.survey_id}"
+        self.access_code = "aMGZtTyFFVhhA0z"
```

Comment:
why is this hardcoded?

Reply:
This shouldn't be hardcoded. To obtain the `access_code `dynamically according to the user, it's necessary to add the user as a participant of the survey, and among its attributes add the anonymous user id. Then, the `get_participant_properties ` method must be consumed filtering by the anonymous user id to obtain the `access_code`.